src/utils/nftRewards.ts

```diff
@@ -237,36 +270,8 @@ export function buildJB721TierParams({
   // `cids` are ordered the same as `rewardTiers` so can get corresponding values from same index
   return cids
     .map((cid, index) => {
-      const contributionFloorWei = parseEther(
-        _rewardTiers[index].contributionFloor.toString(),
-      )
-      const maxSupply = _rewardTiers[index].maxSupply
-      const initialQuantity = BigNumber.from(
-        maxSupply ?? DEFAULT_NFT_MAX_SUPPLY,
-      )
-      const encodedIPFSUri = encodeIpfsUri(cid)
-
-      const reservedRate = _rewardTiers[index].reservedRate
-        ? BigNumber.from(_rewardTiers[index].reservedRate! - 1)
-        : BigNumber.from(0)
-      const reservedTokenBeneficiary =
-        _rewardTiers[index].beneficiary ?? constants.AddressZero
-      const votingUnits = _rewardTiers[index].votingWeight
-        ? BigNumber.from(_rewardTiers[index].votingWeight)
-        : BigNumber.from(0)
-
-      return {
-        contributionFloor: contributionFloorWei,
-        lockedUntil: BigNumber.from(0),
-        initialQuantity,
-        votingUnits,
-        reservedRate,
-        reservedTokenBeneficiary,
-        encodedIPFSUri,
-        allowManualMint: false,
-        shouldUseBeneficiaryAsDefault: false,
-        transfersPausable: false,
-      }
+      const rewardTier = _rewardTiers[index]
+      return nftRewardTierToJB721TierParams(rewardTier, cid)
     })
     .sort((a, b) => {
```

Comment:
**Question:** Wondering if we should have `.slice()` in here for safety, like in `SortNftsByContributionFloor`?

Reply:
done